postgres_exporter/linux/queries_backrest.yml

```diff
@@ -59,6 +59,27 @@ ccp_backrest_last_info:
             description: "Total runtime in seconds of this backup"
 
 
+ccp_backrest_oldest_full_backup:
+    query: "WITH all_backups AS (
+                SELECT config_file
+                 , jsonb_array_elements(data) AS stanza_data
+                FROM monitor.pgbackrest_info(#PGBACKREST_INFO_THROTTLE_MINUTES#)
+            )
+            , per_stanza AS (
+                SELECT config_file
+                 , stanza_data->>'name' AS stanza
+                 , jsonb_array_elements(stanza_data->'backup') AS backup_data
+                FROM all_backups
+            )
+            select 
+		min((backup_data->'timestamp'->>'stop')::bigint)
+            from per_stanza where backup_data->>'type'::text = 'full' group by backup_data->>'type'::text"
+    metrics:
+        - time:
```

Comment:
I was confused why I kept getting the following metric name back: `ccp_backrest_oldest_full_backup_min`

It's because the column being returned by the query doesn't have an alias
```
postgres=# WITH all_backups AS (
                SELECT config_file
                 , jsonb_array_elements(data) AS stanza_data
                FROM monitor.pgbackrest_info(10)
            )
            , per_stanza AS (
                SELECT config_file
                 , stanza_data->>'name' AS stanza
                 , jsonb_array_elements(stanza_data->'backup') AS backup_data
                FROM all_backups
            )
            select 
min((backup_data->'timestamp'->>'stop')::bigint)
            from per_stanza where backup_data->>'type'::text = 'full' group by backup_data->>'type'::text;
    min     
------------
 1617219737
```
Need to make sure your column and metric names match.

Could you also change the column name to be `time_seconds`? It's a common practice to have the units being returned in the metric name and if you look at the other backrest metrics we did that there.

Reply:
This metric also needs to return the `config_file` and `stanza` metrics like the other ones do.

Reply:
Keith is right about the standards.  The units should go on the end of the label.  That both matches other standards and has been catching on all over Crunchy.  I am strongly for it, starting to push that standard upstream to PG itself.  "bytes" goes to the end; "pages" or "blocks" should be bytes; and if everything is bytes, maybe we can drop the unit text altogether..

Reply:
added alias and metric name time_seconds

Reply:
config_file is empty and stanza is db in the container implementation.
To be consistent with others, I will add them as well

Reply:
Done